group-actors/src/main/java/org/sunbird/actors/DeleteGroupActor.java

```diff
@@ -0,0 +1,93 @@
+package org.sunbird.actors;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.actor.core.ActorConfig;
+import org.sunbird.exception.AuthorizationException;
+import org.sunbird.exception.BaseException;
+import org.sunbird.models.MemberResponse;
+import org.sunbird.request.Request;
+import org.sunbird.response.Response;
+import org.sunbird.service.GroupService;
+import org.sunbird.service.GroupServiceImpl;
+import org.sunbird.service.MemberService;
+import org.sunbird.service.MemberServiceImpl;
+import org.sunbird.telemetry.TelemetryEnvKey;
+import org.sunbird.telemetry.util.TelemetryUtil;
+import org.sunbird.util.CacheUtil;
+import org.sunbird.util.GroupRequestHandler;
+import org.sunbird.util.JsonKey;
+import org.sunbird.util.helper.PropertiesCache;
+
+@ActorConfig(
+  tasks = {"deleteGroup"},
+  asyncTasks = {},
+  dispatcher = "group-dispatcher"
+)
+public class DeleteGroupActor extends BaseActor {
+  private CacheUtil cacheUtil = new CacheUtil();
+
+  @Override
+  public void onReceive(Request request) throws Throwable {
+    String operation = request.getOperation();
+    switch (operation) {
+      case "deleteGroup":
+        deleteGroup(request);
+        break;
+      default:
+        onReceiveUnsupportedMessage("DeleteGroupActor");
+    }
+  }
+
+  /**
+   * This method will delete the group
+   *
+   * @param actorMessage
+   * @throws BaseException
+   */
+  private void deleteGroup(Request actorMessage) throws BaseException {
+    logger.info("DeleteGroup method call");
+    GroupRequestHandler requestHandler = new GroupRequestHandler();
+    String userId = requestHandler.getRequestedBy(actorMessage);
+    String groupId = (String) actorMessage.getRequest().get(JsonKey.GROUP_ID);
+    logger.info("Delete group for the groupId {}", groupId);
+
+    if (StringUtils.isEmpty(userId)) {
+      throw new AuthorizationException.NotAuthorized();
+    }
+    GroupService groupService = new GroupServiceImpl();
+
+    Map<String, Object> dbResGroup = groupService.readGroup(groupId);
+    // Only Group Creator should be able to delete the group
+    if (!userId.equals((String) dbResGroup.get(JsonKey.CREATE_BY))) {
+      throw new AuthorizationException.NotAuthorized();
+    }
+
+    // Get all members belong to the group
+    MemberService memberService = new MemberServiceImpl();
+    List<MemberResponse> membersInDB = memberService.fetchMembersByGroupId(groupId);
+    Response response = groupService.deleteGroup(groupId, membersInDB);
+    // delete cache for the group and all members belong to the group
+    boolean isUseridRedisEnabled =
+        Boolean.parseBoolean(
+            PropertiesCache.getInstance().getConfigValue(JsonKey.ENABLE_USERID_REDIS_CACHE));
+    if (isUseridRedisEnabled) {
```

Comment:
Add the code to remove groupId_member cache

Reply:
updated